### PR TITLE
send 100 peers to connected peers

### LINKED ISSUE #3832: Peers network tester
- [x] Allowing the configuration of the number of threads for testing the peers. Each thread will listen to a common channel to launch their tests. If no message in 2 seconds they will test a peer that hasn't been tested for long time (can use index_by_newest)
- [x] Randomly selecting 100 peers to send to another peer. The selected peers must have been online within the last 3 days.
- [x]  Send the selected peers after a normal handshake succeed or failed
- [x]  Send 100 random selected peers each 10 seconds

### PR DESCRIPTION
**Pull request by modship**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### Summary
🔄📡🎲

This pull request refactors and improves the peer management and network communication logic of the protocol worker. It introduces a shared `PeerDB` for the connectivity module and the peer handler, and a new channel for peer messages. It also adds periodic peer list exchange, better peer selection, and more efficient and readable testing code.

### Walkthrough
*  Move peer management handler code from `handlers/peer_handler/mod.rs` to `connectivity.rs` and add new imports for crossbeam and peer types ([link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL1-R3), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eR1), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eR21))
*  Add shared peer database and message channels as arguments to connectivity module and peer management handler to avoid duplication and enable communication ([link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eL66-R68), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eR74-R75), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eL78-R87), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-c2d72e603a7e395a81a361e7ff5aa497652841761cc6dc0c1f587abf2d778b7eR98-R106), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL58-R78), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL157-R180))
*  Implement periodic sending of peer lists to random peers using tick function, peer database selection method, and active connections ([link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL9-R12), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL80-R104), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-92fd577f9eff16de739adf93ad3e0bb0a0993f097a5f87d96e58666eeb704482L5-R7), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-92fd577f9eff16de739adf93ad3e0bb0a0993f097a5f87d96e58666eeb704482R17-R18), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-92fd577f9eff16de739adf93ad3e0bb0a0993f097a5f87d96e58666eeb704482R98-R134))
*  Update message deserializer and serializer to include peer management message types and use message channels ([link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL100-R123), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-7165d31835e99b9c8715c9c94afb5149d2e6b8c39cb3a3dd244811f71c1ae65cL150-R173))
*  Remove unnecessary clone of peer_id in tester and add comment to document new method ([link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-9d9bbaebe84221f667ce39359894963e4c0212978110bdc7bc16fff3b7d69e09L139-R141), [link](https://github.com/massalabs/massa/pull/3864/files?diff=unified&w=0#diff-9d9bbaebe84221f667ce39359894963e4c0212978110bdc7bc16fff3b7d69e09R172))

### CODE REVIEW

File: massa-protocol-worker-2/src/handlers/peer_handler/mod.rs [hunk 1]
```diff
@@ -97,7 +119,7 @@ impl PeerManagementHandler {
                             }
                            }
                         },
-                        recv(receiver) -> msg => {
+                        recv(receiver_msg) -> msg => {
                             let (peer_id, message_id, message) = match msg {
                                 Ok((peer_id, message_id, message)) => (peer_id, message_id, message),
```
Comment (AurelienFT) [inline massa-protocol-worker-2/src/handlers/peer_handler/mod.rs:124]: Here you should deserialize the message (you can take example in other handlers) and manage it

Comment (modship) [reply massa-protocol-worker-2/src/handlers/peer_handler/mod.rs:124]: it's done on line 139 ?!

Comment (AurelienFT) [reply massa-protocol-worker-2/src/handlers/peer_handler/mod.rs:124]: Ah yes ok I'm dumb

**Review by AurelienFT — APPROVED** [review]

**Comment by AurelienFT** [comment]
Merging it now myself because I need to merge this changes in my new PR.

**Pull request merged**